compiler/z/codegen/OMRMachine.cpp

```diff
@@ -6444,487 +6444,252 @@ OMR::Z::Machine::initializeGlobalRegisterTable()
-   _globalRegisterNumberToRealRegisterMap[6] = TR::RealRegister::GPR9;     // preserved
-   _globalRegisterNumberToRealRegisterMap[7] = TR::RealRegister::GPR10;    // preserved
-   _globalRegisterNumberToRealRegisterMap[8] = TR::RealRegister::GPR11;    // preserved -- non-Java may lock
-   _globalRegisterNumberToRealRegisterMap[9] = TR::RealRegister::GPR12;    // preserved -- non-Java may lock
-
-   // Access Registers
-   self()->setFirstGlobalAccessRegisterNumber(10);
-   _globalRegisterNumberToRealRegisterMap[10] = (uint32_t) (-1);              // TR::RealRegister::AR0; locked on zLinux
-   _globalRegisterNumberToRealRegisterMap[11] = (uint32_t) (-1);              // TR::RealRegister::AR1; locked on zLinux
-   // /// /// Disable and test on only on ZOS where the other ARs are protected by system linkage
-   _globalRegisterNumberToRealRegisterMap[12] = (uint32_t) (-1);              // TR::RealRegister::AR2;
-   _globalRegisterNumberToRealRegisterMap[13] = (uint32_t) (-1);              // TR::RealRegister::AR3;
-   _globalRegisterNumberToRealRegisterMap[14] = (uint32_t) (-1);              // TR::RealRegister::AR4;
-   _globalRegisterNumberToRealRegisterMap[15] = (uint32_t) (-1);              // TR::RealRegister::AR5;
-   _globalRegisterNumberToRealRegisterMap[16] = (uint32_t) (-1);              // TR::RealRegister::AR6;
-   _globalRegisterNumberToRealRegisterMap[17] = (uint32_t) (-1);              // TR::RealRegister::AR7;
-   // /// ///
-
-   _globalRegisterNumberToRealRegisterMap[18] = TR::RealRegister::AR8;      // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[19] = (uint32_t) (-1);               // TR::RealRegister::AR9;      // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[20] = (uint32_t) (-1);               // TR::RealRegister::AR10;     // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[21] = (uint32_t) (-1);               // TR::RealRegister::AR11;     // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[22] = (uint32_t) (-1);               // TR::RealRegister::AR12;     // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[23] = (uint32_t) (-1);               // TR::RealRegister::AR13;     // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[24] = (uint32_t) (-1);               // TR::RealRegister::AR14;     // preserved ZOS
-   _globalRegisterNumberToRealRegisterMap[25] = (uint32_t) (-1);               // TR::RealRegister::AR15;     // preserved ZOS
-   self()->setLastGlobalAccessRegisterNumber(25);
+   self()->setLastLinkageFPR(p-1);
 
-   self()->setLastGlobalGPRRegisterNumber(25);        // Index of last global GPR
-   self()->setLast8BitGlobalGPRRegisterNumber(25);    // Index of last global 8bit Reg
-
-   // additional (forced) restricted regs.
-   // Similar code in TR::S390PrivateLinkage::initS390RealRegisterLinkage() for RA
-
-   for (int32_t i = self()->getFirstGlobalGPRRegisterNumber(); i < self()->getLastGlobalGPRRegisterNumber(); ++i)
+   // Preserved regs, vmthread last
```

Comment:
Looks like this comment is obsolete now as vmThread is added to register file earlier in this function (Line 6545)

Reply:
You're right. Fixed in a686991